model/flow/identifier.go

```diff
@@ -64,6 +66,16 @@ func (id Identifier) Format(state fmt.State, verb rune) {
 	}
 }
 
+// IsSampled is a utility method to sample entities based on their ids,
+// percentage adjust how often we sample the entity.
+// 0 means 0 data will be collected.
+// 50 means 50% of the entities would be collected.
+// 100 means all data will be collected.
+func (id Identifier) IsSampled(percentage uint) bool {
+	// take the first 8 bytes and check the first few bits based on the percentage
+	return binary.BigEndian.Uint64(id[:8])>>(64-int(math.Min(float64(percentage)/float64(1.5), 64))) == 0
```

Comment:
Not sure why we need our own way of randomness?

Reply:
cause we wrap the open tracer inside the tracer module, and when we reach to the tracer sampling it is already late and we have done alot of extra work